homeassistant/components/duotecno/climate.py

```diff
@@ -0,0 +1,79 @@
+"""Support for Duotecno climate devices."""
+from typing import Any
+
+from duotecno.unit import SensUnit
+
+from homeassistant.components.climate import (
+    ClimateEntity,
+    ClimateEntityFeature,
+    HVACMode,
+)
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import ATTR_TEMPERATURE, UnitOfTemperature
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+
+from .const import DOMAIN, HVACMODE, PRESETMODES
+from .entity import DuotecnoEntity, api_call
+
+
+async def async_setup_entry(
+    hass: HomeAssistant,
+    entry: ConfigEntry,
+    async_add_entities: AddEntitiesCallback,
+) -> None:
+    """Set up Duotecno climate based on config_entry."""
+    cntrl = hass.data[DOMAIN][entry.entry_id]
+    async_add_entities(
+        DuotecnoClimate(channel) for channel in cntrl.get_units(["SensUnit"])
+    )
+
+
+class DuotecnoClimate(DuotecnoEntity, ClimateEntity):
+    """Representation of a Duotecno climate entity."""
+
+    _unit: SensUnit
+    _attr_supported_features = (
+        ClimateEntityFeature.TARGET_TEMPERATURE | ClimateEntityFeature.PRESET_MODE
+    )
+    _attr_temperature_unit = UnitOfTemperature.CELSIUS
+    _attr_hvac_modes = [HVACMode.OFF, HVACMode.HEAT, HVACMode.COOL]
```

Comment:
You could also create that reverse map for hvac_modes to enable you to use `list(HVAC_MODES)`. In this case you never do a quick lookup, but you are assured that every hvac mode you support, has a duotecno state in front of it.

Isn't a blocker, but would be nice stylewise